Samples/HelloWorld/OrleansHost/OrleansHostWrapper.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System;
+using System.Net;
+
+using Orleans.Runtime.Configuration;
+using Orleans.Runtime.Host;
+
+namespace OrleansHost
+{
+    class OrleansHostWrapper
+    {
+        private SiloHost siloHost;
+
+        public OrleansHostWrapper(string[] args)
+        {
+            SiloArgs siloArgs = ParseArguments(args);
+            if (siloArgs != null)
+            {
+                Init(siloArgs);
+            }
+        }
+
+        private void Init(SiloArgs siloArgs)
+        {
+            var config = ClusterConfiguration.LocalhostPrimarySilo();
+            config.Globals.DeploymentId = siloArgs.DeploymentId;
+            config.AddMemoryStorageProvider();
+            siloHost = new SiloHost(siloArgs.SiloName, config);
+            siloHost.LoadOrleansConfig();
+        }
+
+        public int Run()
+        {
+            //if siloHost is not constructed yet due to wrong cmd param
+            if (siloHost == null)
+            {
+                Console.Error.WriteLine("Failed to initialize Orleans silo due to bad command line arguments");
+                return 1;
+            }
+            else
+            {
+                try
+                {
+                    siloHost.InitializeOrleansSilo();
+
+                    if (siloHost.StartOrleansSilo())
+                    {
+                        Console.WriteLine("Successfully started Orleans silo '{0}' as a {1} node.", siloHost.Name, siloHost.Type);
+                        return 0;
+                    }
+                    else
+                    {
+                        throw new SystemException(string.Format("Failed to start Orleans silo '{0}' as a {1} node.", siloHost.Name, siloHost.Type));
```

Comment:
avoid using SystemException as it is deprecated. You might use OrleansException instead, as it seems appropriate here